### PR TITLE
Sam tov zn vis

### PR DESCRIPTION
**Pull request by SamTov**
Make visualizer an optional add-on that requires installing the ZnVis package separately. At the same time, remove the SDF code from the package using comments to avoid loss of features.

Has been tested on a fresh environment without ZnVis package and it raises the required logging statement and prevent the visualizer from being called.

### CODE REVIEW

File: .github/workflows/pytest.yml [hunk 1]
```diff
@@ -35,6 +35,7 @@ jobs:
         run: |
           python -m pip install --upgrade pip
           pip install -r dev-requirements.txt
+          pip install znvis
```
Comment (PythonFZ) [inline .github/workflows/pytest.yml]: Why not put it into dev-requirements.txt?

**Comment by PythonFZ** [comment]
I think we should merge this with  #467 I don't think we need to branches for that.

**Comment by SamTov** [comment]
> I think we should merge this with #467 I don't think we need to branches for that.

Yep but because a lot of vis stuff was removed in #467 and it is probably easier not to go back and re-add it we can probably just close that and use this PR.

**Comment by PythonFZ** [comment]
> > I think we should merge this with #467 I don't think we need to branches for that.
> 
> Yep but because a lot of vis stuff was removed in #467 and it is probably easier not to go back and re-add it we can probably just close that and use this PR.

There are some parts in the other PR that weren't used in MDSuite anymore so I would like to merge them.

**Comment by SamTov** [comment]
> > > I think we should merge this with #467 I don't think we need to branches for that.
> > 
> > 
> > Yep but because a lot of vis stuff was removed in #467 and it is probably easier not to go back and re-add it we can probably just close that and use this PR.
> 
> There are some parts in the other PR that weren't used in MDSuite anymore so I would like to merge them.

Yep that's no problem for me just have to be careful that none of the stuff is removed. I'd like to keep basically everything related to the SDF just have it commented out.

**Review by PythonFZ — APPROVED** [review]
Works in Binder - looks good!

File: .github/workflows/pytest.yml (hunk 1 above)
Comment (SamTov) [reply .github/workflows/pytest.yml]: Thought about it, and don't mind the idea, I left it out as I figured it is not explicitly a dev requirement. But if you think it also can go there then I can move it.

Comment (PythonFZ) [reply .github/workflows/pytest.yml]: I would put it there, because it is required for the tests to run. If you run them locally they will fail otherwise.

Comment (SamTov) [reply .github/workflows/pytest.yml]: solved in 57b988f

**Pull request merged**